framework/web/Request.php

```diff
@@ -195,29 +227,61 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.12
+     * @return array The filtered list of headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
+        // Don't use getUserIP here since that depends on using headers.
+        $ip = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : null;
+        foreach($this->trustedHostConfig as $hostRegex => $trustedHeaders) {
+            if (is_numeric($hostRegex)) {
```

Comment:
`is_numeric()` can return `true` for some intranet hosts. I suggest use `is_array($trustedHeaders)`?